halo2_proofs/src/plonk.rs

```diff
@@ -44,14 +44,12 @@ use std::io;
 pub fn vk_read<C: SerdeCurveAffine, R: io::Read, ConcreteCircuit: Circuit<C::Scalar>>(
     reader: &mut R,
     format: SerdeFormat,
-    compress_selectors: bool,
     #[cfg(feature = "circuit-params")] params: ConcreteCircuit::Params,
 ) -> io::Result<VerifyingKey<C>>
 where
     C::Scalar: SerdePrimeField + FromUniformBytes<64>,
 {
-    let (_, cs, _) = compile_circuit_cs::<_, ConcreteCircuit>(
-        compress_selectors,
+    let (_, cs) = compile_circuit_cs::<_, ConcreteCircuit>(
```

Comment:
There's a detail here that I don't understand.  With this change the `cs` we get here has selector columns (they have not yet been converted to fixed columns).

Then afterwards we convert the `ConstraintSystem` into `ConstraintSystemMid`.  In that conversion we're converting `Expression` into `ExpressionMid`, and if the `Expression` has a selector it will panic: https://github.com/privacy-scaling-explorations/halo2/blob/9eedeb5d6eb9a119d34307697e670b0bb043a5a5/halo2_frontend/src/plonk/circuit/expression.rs#L471

So I think that calling `vk_read` or `pk_read` for a circuit that uses `Selector`s will panic.

Reply:
Hmm, I don't think so.
The `pk_read` & `vk_read` are utilities to read the keys that were already compiled & stored somewhere.
In other word, they include the info of `fixed` columns converted from `Selector`s, not `Selector`s.

Plus, the process of converting the `Selector`s to `fixed` columns needs the synthesized circuit, not just circuit configuration.